data/test/macros/wml_unit_test_macros.cfg

```diff
@@ -171,9 +171,33 @@ bob#endarg
 #arg COMPARE
 numerical_equals#endarg
 
+#arg DEATH_RESULT
+FAIL#endarg
+
     [event]
         name = start
 
+        # allow specifying a different number of strikes per weapon
+        [set_variables]
+            name=strikes
```

Comment:
There's two conflicting design choices in this function - most of the variables are using `{ID1}_{ID2}` to disambiguate multiple `CHECK_STRIKES` instances being used in a single test. This isn't, and yet the tests still pass.

Reply:
The documentation for `[split]` says:
> splits a textual list into an array which will then be set to data

which I take to mean (which the tests reflect) that existing data in the variable is replaced. Otherwise yeah, if it were appended then the tests would fail when an error is logged about trying to attack with a weapon that doesn't exist.

Reply:
> which I take to mean (which the tests reflect) that existing data in the variable is replaced.

That's incorrect. Whether the existing data is replaced depends on the `mode=` key. However, I _think_ the default mode is `replace`, which does indeed clear the variable before adding data to it. It would be appended if you had `mode=append`, which you don't. The wiki seems to be missing documentation of the default mode however, so I'm not 100% sure of this; you'll need to check the code if you want to be certain.

Reply:
Looks like it's the default:
https://github.com/wesnoth/wesnoth/blob/28a895985499afac8809c4c03ed080f4fe133da5/src/game_events/action_wml.cpp#L712-L736

Reply:
My point was about the names of the variables, and the assumptions in them when a test instanciates the `CHECK_STRIKES` macro multiple times. However, I now see it predates this PR, so think the right resolution would be to document that "The implementation currently only supports one CHECK_STRIKES per unique ID1 per unit test".

Based on their names:
* `strikes` suggests all of the actions of the current `CHECK_STRIKES` will be handled before anything else access that variable
* `{ID1}_{ID2}_count` suggests that it needs to be keyed to the specific pair of units, even though it's set to zero at the same time as `strikes`, and accessed during the same timespan as `strikes`
* the filters on the events themselves depend only on `ID1`, thus the `{ID1}_{ID2}_count` is incremented when `ID1` attacks, regardless of the target

Reply:
Added the doc comment and a couple tests of it.